src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramParser.java

```diff
@@ -46,6 +46,12 @@
     static final ParseField OFFSET = new ParseField("offset");
     static final ParseField PRE_OFFSET = new ParseField("", "pre_offset");
     static final ParseField POST_OFFSET = new ParseField("", "post_offset");
+    static final ParseField PRE_ZONE = new ParseField("", "pre_zone");
+    static final ParseField POST_ZONE = new ParseField("", "post_zone");
```

Comment:
You mean like `new ParseField("pre_zone").withAllDeprecated("")` ? Since the argument of withAllDeprecated seems to replace the original names this would have to be empty. As long as having the deprecated name as preferedName still raises the desired exception, that would work, right?